diff --git a/core.cjs b/core.cjs
index ba878b22d7f1f1b237c75b0399555ea67ef6d8ab..03f7b0541bf2e395e65cfd37bf17ccbc69c03599 100644
--- a/core.cjs
+++ b/core.cjs
@@ -248,9 +248,7 @@ function replaceEach(selector, iteratee) {
 function replaceReference(selector, reference) {
     return selector.replace(/&/g, reference);
 }
-const collator = new Intl.Collator('en', {
-    numeric: true
-});
+
 /** The calculated precedence taking all variants into account. */ /* The precedence of the properties within {@link d}. */ /** The name to use for `&` expansion in selectors. Maybe empty for at-rules like `@import`, `@font-face`, `@media`, ... */ /**
  * Find the array index of where to add an element to keep it sorted.
  *
@@ -267,7 +265,7 @@ const collator = new Intl.Collator('en', {
 function compareTwindRules(a, b) {
     // base and overrides (css) layers are kept in order they are declared
     let layer = a.p & Layer.o;
-    return layer == (b.p & Layer.o) && (layer == Layer.b || layer == Layer.o) ? 0 : a.p - b.p || a.o - b.o || collator.compare(byModifier(a.n), byModifier(b.n)) || collator.compare(byName(a.n), byName(b.n));
+    return layer == (b.p & Layer.o) && (layer == Layer.b || layer == Layer.o) ? 0 : a.p - b.p || a.o - b.o;
 }
 function byModifier(s) {
     return (s || '').split(/:/).pop().split('/').pop() || '\x00';
diff --git a/core.js b/core.js
index 26018b3a0b42087e0458f21035720a9598f48747..9d6d759c0afbcd27fdd221728c83e15de9855905 100644
--- a/core.js
+++ b/core.js
@@ -239,9 +239,7 @@ function replaceEach(selector, iteratee) {
 function replaceReference(selector, reference) {
     return selector.replace(/&/g, reference);
 }
-let collator = new Intl.Collator('en', {
-    numeric: true
-});
+
 /** The calculated precedence taking all variants into account. */ /* The precedence of the properties within {@link d}. */ /** The name to use for `&` expansion in selectors. Maybe empty for at-rules like `@import`, `@font-face`, `@media`, ... */ /**
  * Find the array index of where to add an element to keep it sorted.
  *
@@ -258,7 +256,7 @@ let collator = new Intl.Collator('en', {
 function compareTwindRules(a, b) {
     // base and overrides (css) layers are kept in order they are declared
     let layer = a.p & Layer.o;
-    return layer == (b.p & Layer.o) && (layer == Layer.b || layer == Layer.o) ? 0 : a.p - b.p || a.o - b.o || collator.compare(byModifier(a.n), byModifier(b.n)) || collator.compare(byName(a.n), byName(b.n));
+    return layer == (b.p & Layer.o) && (layer == Layer.b || layer == Layer.o) ? 0 : a.p - b.p || a.o - b.o;
 }
 function byModifier(s) {
     return (s || '').split(/:/).pop().split('/').pop() || '\x00';
