diff --git a/lib/scope/index.js b/lib/scope/index.js
index f0c9748c9dcb462ec420fad7d6f673f59bad2053..25f1bc8632f1b64e980bcd1cae31f26dd0b03cd1 100644
--- a/lib/scope/index.js
+++ b/lib/scope/index.js
@@ -7,7 +7,6 @@ exports.default = void 0;
 var _renamer = require("./lib/renamer.js");
 var _index = require("../index.js");
 var _binding = require("./binding.js");
-var _globals = require("globals");
 var _t = require("@babel/types");
 var t = _t;
 var _cache = require("../cache.js");
@@ -956,7 +955,71 @@ class Scope {
   }
 }
 exports.default = Scope;
-Scope.globals = Object.keys(_globals.builtin);
+Scope.globals = Object.keys({
+  Array: false,
+  ArrayBuffer: false,
+  Atomics: false,
+  BigInt: false,
+  BigInt64Array: false,
+  BigUint64Array: false,
+  Boolean: false,
+  constructor: false,
+  DataView: false,
+  Date: false,
+  decodeURI: false,
+  decodeURIComponent: false,
+  encodeURI: false,
+  encodeURIComponent: false,
+  Error: false,
+  escape: false,
+  eval: false,
+  EvalError: false,
+  Float32Array: false,
+  Float64Array: false,
+  Function: false,
+  globalThis: false,
+  hasOwnProperty: false,
+  Infinity: false,
+  Int16Array: false,
+  Int32Array: false,
+  Int8Array: false,
+  isFinite: false,
+  isNaN: false,
+  isPrototypeOf: false,
+  JSON: false,
+  Map: false,
+  Math: false,
+  NaN: false,
+  Number: false,
+  Object: false,
+  parseFloat: false,
+  parseInt: false,
+  Promise: false,
+  propertyIsEnumerable: false,
+  Proxy: false,
+  RangeError: false,
+  ReferenceError: false,
+  Reflect: false,
+  RegExp: false,
+  Set: false,
+  SharedArrayBuffer: false,
+  String: false,
+  Symbol: false,
+  SyntaxError: false,
+  toLocaleString: false,
+  toString: false,
+  TypeError: false,
+  Uint16Array: false,
+  Uint32Array: false,
+  Uint8Array: false,
+  Uint8ClampedArray: false,
+  undefined: false,
+  unescape: false,
+  URIError: false,
+  valueOf: false,
+  WeakMap: false,
+  WeakSet: false,
+});
 Scope.contextVariables = ["arguments", "undefined", "Infinity", "NaN"];
 
 //# sourceMappingURL=index.js.map
